src/internal/analytics-metadata/__tests__/labels-utils.test.tsx

```diff
@@ -0,0 +1,228 @@
+          content
+        </div>
+      </div>
+    );
+    const target = container.querySelector('#target');
+    expect(getLabelFromElement(target as HTMLElement)).toEqual('');
+  });
+});
+
+describe('processLabel', () => {
+  test('returns an empty string when element and/or identifier are null', () => {
+    const { container } = render(
+      <div>
+        <div id="target">content</div>
+      </div>
+    );
+    const target = container.querySelector('#target') as HTMLElement;
+
+    expect(processLabel(null, null)).toEqual('');
+    expect(processLabel(target, null)).toEqual('');
+    expect(processLabel(null, 'selector')).toEqual('');
+  });
+  test('returns an empty string when the selector cannot be resolved to an element', () => {
+    const { container } = render(
+      <div>
+        <div id="target">content</div>
+      </div>
+    );
+    const target = container.querySelector('#target') as HTMLElement;
+
+    expect(processLabel(target, 'invalid-selector')).toEqual('');
+  });
+  test('returns node label when selector is an empty string', () => {
+    const { container } = render(
+      <div>
+        <div id="target">content</div>
+      </div>
+    );
+    const target = container.querySelector('#target') as HTMLElement;
+
+    expect(processLabel(target, '')).toEqual('content');
+    expect(processLabel(target, { selector: '' })).toEqual('content');
+  });
+  test('returns node label when selector is an empty string', () => {
+    const { container } = render(
+      <div>
+        <div id="target">
+          <div>content</div>
+          <div className="label-class">
+            <div>second content</div>
+            <h1>label to return</h1>
+          </div>
+        </div>
+      </div>
+    );
+    const target = container.querySelector('#target') as HTMLElement;
+
+    expect(processLabel(target, '.label-class h1')).toEqual('label to return');
+    expect(processLabel(target, { selector: '.label-class h1' })).toEqual('label to return');
+  });
+  test('returns first available label when selector is an array', () => {
+    const { container } = render(
+      <div>
+        <div id="target">
+          <div>content</div>
+          <div className="label-class">
+            <div>second content</div>
+            <h1>heading1</h1>
+            <h2></h2>
+            <h3>heading3</h3>
+          </div>
+        </div>
+      </div>
+    );
+    const target = container.querySelector('#target') as HTMLElement;
+
+    expect(processLabel(target, { selector: ['.label-class h1', '.label-class h2', '.label-class h3'] })).toEqual(
+      'heading1'
+    );
+    expect(
+      processLabel(target, { selector: ['.label-wrong-class h1', '.label-wrong-class h2', '.label-class h3'] })
+    ).toEqual('heading3');
+    expect(processLabel(target, { selector: ['.label-class h2', '.label-class h3'] })).toEqual('heading3');
+  });
+  test('respects the root property', () => {
```

Comment:
With the current test name I don't think its clear what the root property is supposed to do. Can we reword this test name to what the root property should do when set to a specific value?

Reply:
Added comments